scheds/rust/scx_layered/src/main.rs

```diff
@@ -1030,16 +1048,57 @@ impl Layer {
             }
         }
 
+        let mut nodes = topo.nodes().iter().collect::<Vec<_>>().clone();
+        let num_nodes = nodes.len();
+        let is_left = idx % 2 == 0;
+        if is_left {
+            nodes.rotate_left(if idx <= num_nodes { idx } else { idx % num_nodes });
```

Comment:
Maybe sth like `let rot_by |idx, len| if idx <= len { idx } else { idx % len };` would help? Hmm... isn't this the same as `idx % num_nodes`?